src/_path.h

```diff
@@ -834,6 +834,17 @@ inline bool segments_intersect(const double &x1,
     // it looks the atol value matters here bacause of round-off errors
     const double rtol = 1e-10;
     const double atol = 1e-13;
+
+    // if either segment is 0 length, they do not intersect
+    // length-squared of each segment
+    const double lensq_A = (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2);
+    const double lenqs_B = (x3 - x4) * (x3 - x4) + (y3 - y4) * (y3 - y4);
+
+    // one of the segments is 0 length
+    if (isclose(lensq_A, 0, rtol, atol) || isclose(lenqs_B, 0, rtol, atol)) {
```

Comment:
Do we really need `isclose`? If both end points are identical, `lensq` will be exactly 0. Or can it be that the points themselves are calculated and may have a numeric uncertainty.

Reply:
I'm worried about cases where they are computed and almost, but not identically equal (like `.3` vs `3 * .1`).   Because we use `isclose` below to determine if the determinant is 0 so we need to use it here as well.

Reply:
Fair enough.